plugin/src/main/java/org/opensearch/ml/indices/MLIndicesHandler.java

```diff
@@ -97,12 +98,23 @@ public void initMLIndexIfAbsent(MLIndex index, ActionListener<Boolean> listener)
                                 .putMapping(
                                     new PutMappingRequest().indices(indexName).source(mapping, XContentType.JSON),
                                     ActionListener.wrap(response -> {
-                                        if (response.isAcknowledged()) {
-                                            indexMappingUpdated.get(indexName).set(true);
-                                            internalListener.onResponse(true);
-                                        } else {
-                                            internalListener.onFailure(new MLException("Failed to update index: " + indexName));
-                                        }
+                                        UpdateSettingsRequest updateSettingRequest = new UpdateSettingsRequest();
```

Comment:
I see you removed `if (response.isAcknowledged()) {`, I think we should keep this checking

Reply:
In line 107

Reply:
Or you think we need to check twice?

Reply:
If index mapping not sucessfull, no need to update setting, I think we should twice